From 3e7c56f0db15326d31bb905daffb82f5c7137962 Mon Sep 17 00:00:00 2001
From: Sebastian Lohff <sebastian.lohff@sap.com>
Date: Fri, 16 May 2025 18:00:25 +0200
Subject: [PATCH] Set service type when registering provider config

When registering the provider configuration with the service type
manager we need to specify a service type for ProviderConfiguration(),
as described in the Neutron Contributors documentation[0]. If this is
not done it might lead to duplicate service providers configuration, as
the ProviderConfiguration no longer filters out other services.

[0] https://docs.openstack.org/neutron/latest/contributor/contribute.html#service-providers

Upstream-master: https://review.opendev.org/c/openstack/neutron-fwaas/+/950202
Resolves: #OSPRH-18582

Change-Id: I33ba7fdf1748b1604f3ed72a3257d6cf697d6e2b
(cherry picked from commit f7f42f2d9b15c8a97e43116af7ff713650616fd7)
---
 neutron_fwaas/services/firewall/fwaas_plugin_v2.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/neutron_fwaas/services/firewall/fwaas_plugin_v2.py b/neutron_fwaas/services/firewall/fwaas_plugin_v2.py
index 7eb8260ec..8be2b5545 100644
--- a/neutron_fwaas/services/firewall/fwaas_plugin_v2.py
+++ b/neutron_fwaas/services/firewall/fwaas_plugin_v2.py
@@ -54,7 +54,8 @@ class FirewallPluginV2(Firewallv2PluginBase):
         service_type_manager = st_db.ServiceTypeManager.get_instance()
         service_type_manager.add_provider_configuration(
             fwaas_constants.FIREWALL_V2,
-            provider_conf.ProviderConfiguration('neutron_fwaas'))
+            provider_conf.ProviderConfiguration('neutron_fwaas',
+                                                fwaas_constants.FIREWALL_V2))
 
         # Load the default driver
         drivers, default_provider = service_base.load_drivers(
-- 
2.50.1

